turn_tracker.gd

```diff
@@ -15,21 +19,59 @@ func _ready() -> void:
 	]
 	setTurnOrder(turnOrder)
 
+func _process(delta: float) -> void:
+	currentTurnTaker = Global.turnOrder[turnIndex]
 
 func resetTurnOrder() -> void:
 	Global.turnOrder = []
 
 func setTurnOrder(newOrder: Array) -> void:
 	Global.turnOrder = newOrder
 
+func resetNumberOfUnits() -> void:
+	numberOfUnits = 0
+
+func addUnits(count: int = 1) -> void:
+	numberOfUnits += count
+
 func subtractUnits(count: int) -> void:
 	numberOfUnits -= count
```

Comment:
we don't need an add and subtract unit tbh since we can just pass in negatives, but this will be easier to read sooooo